index.js

```diff
@@ -8,6 +8,78 @@ function isString(s) {
   return 'string' === typeof s
 }
 
+var hasLink = exports.hasLink =
+  function (data) {
+    return /(@|%|&)[A-Za-z0-9\/+]{43}=\.[\w\d]+$/.exec(data)
+  }
+
+var hasFeedId = exports.hasFeed = exports.hasFeedId =
+  function (data) {
+    return /(@[A-Za-z0-9\/+]{43}=\.(?:sha256|ed25519))/.exec(data)
+  }
+
+var hasMsgId = exports.hasMsg = exports.hasMsgId =
+  function (data) {
+    return /(%[A-Za-z0-9\/+]{43}=\.sha256)/.exec(data)
+  }
+
+var hasBlobId = exports.hasBlob = exports.hasBlobId =
+  function (data) {
+    return /(&[A-Za-z0-9\/+]{43}=\.sha256)/.exec(data)
+  }
+
+var hasTagId = exports.hasTag = exports.hasTagId =
```

Comment:
The tag functions presuppose we're going to introduce tags. If they stay in, this PR will have to wait for us to decide on tags.